### PR TITLE
 dfc 3460 bad request  handling from the av api

### PR DESCRIPTION
**Pull request by dineshtripathi**
Handling Bad Request from the AV API as some of the framework / standard throwing exception while fetching the API data for AV Feed

### CODE REVIEW

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Function.GetAVForSoc/ApprenticeshipVacancyFunction.cs [hunk 1]
```diff
@@ -31,8 +33,9 @@ public MappedVacancySummary GetOutput()
             {
                 Vacancies = vacancySummaries,
                 SocCode = socMapping.SocCode,
-                SocMappingId = socMapping.SocMappingId,
-                AccessToken = socMapping.AccessToken
+                SocMappingId =  socMapping.SocMappingId,
+                AccessToken = (vacancySummaries.Any()) ? socMapping.AccessToken:null,
+                IsValidVacancy = (vacancySummaries.Any()) ?true:false
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Integration.AVFeed.Function.GetAVForSoc/ApprenticeshipVacancyFunction.cs]: Not correct logic, there will valid mapping but returns 0 vacancies at the time of call and can have vacancy added to it later etc. So checking for summary count as 0 is not a valid logic.

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Data/Models/MappedVacancySummary.cs [hunk 2]
```diff
@@ -7,8 +7,8 @@ public class MappedVacancySummary : BaseIntegrationModel
     {
         public string SocCode { get; set; }
         public Guid SocMappingId { get; set; }
-
         public IEnumerable<ApprenticeshipVacancySummary> Vacancies { get; set; }
         public string AccessToken { get; set; }
+        public bool IsValidVacancy { get; set; }
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Integration.AVFeed.Data/Models/MappedVacancySummary.cs]: Change this to "IsValidSocMapping"

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs [hunk 3]
```diff
@@ -83,12 +92,12 @@ public AVAPIService(IApprenticeshipVacancyApi apprenticeshipVacancyApi, IApplica
...
 
             logger.Trace($"Extracting AV summaries for SOC {mapping.SocCode} page : {pageNumber}");
-           
+
             var queryString = HttpUtility.ParseQueryString(string.Empty);
 
-            queryString["standardLarsCodes"] = string.Join(",", mapping.Standards);
-            queryString["frameworkLarsCodes"] = string.Join(",", mapping.Frameworks);
-            queryString["pageSize"] = ConfigurationManager.AppSettings.Get("FAA.PageSize");;
+            queryString["standardLarsCodes"] = string.Join(",", mapping.Standards.Where(std=>!string.IsNullOrEmpty(std)));
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs]: would be nice if you could following the styling guidelines and add spaces where needed to make it readable. :see_no_evil:

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs [hunk 4]
```diff
@@ -36,40 +36,38 @@ public AVAPIService(IApprenticeshipVacancyApi apprenticeshipVacancyApi, IApplica
...
             {
-                var apprenticeshipVacancySummaryResponse = await GetAVSumaryPageAsync(mapping, ++pageNumber);
+                    var apprenticeshipVacancySummaryResponse = await GetAVSumaryPageAsync(mapping, ++pageNumber);
 
-                logger.Trace($"Got {apprenticeshipVacancySummaryResponse.TotalReturned} vacancies of {apprenticeshipVacancySummaryResponse.TotalMatched} on page: {pageNumber} of {apprenticeshipVacancySummaryResponse.TotalPages}");
+                    logger.Trace(
+                        $"Got {apprenticeshipVacancySummaryResponse.TotalReturned} vacancies of {apprenticeshipVacancySummaryResponse.TotalMatched} on page: {pageNumber} of {apprenticeshipVacancySummaryResponse.TotalPages}");
 
-                avSummary.AddRange(apprenticeshipVacancySummaryResponse.Results);
+                    avSummary.AddRange(apprenticeshipVacancySummaryResponse.Results);
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Integration.AVFeed.Service.AVAPI/AVAPIService.cs]: This variable may be null as you are returning null in GetAVSumaryPageAsync method, but no null checks, clearly no unit tests either, please can you update.

File: DFC.Integration.AVFeed/DFC.Integration.AVFeed.Core/Extensions/HttpResponseExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+            var content = await response.Content.ReadAsStringAsync();
+
+            response.Content?.Dispose();
+
+            throw new AvApiResponseException(response.StatusCode, content);
+        }
+    }
+
+    public class AvApiResponseException : Exception
```
Comment (Muthuramana) [inline DFC.Integration.AVFeed/DFC.Integration.AVFeed.Core/Extensions/HttpResponseExtensions.cs]: Move this to a new file

**Comment by dineshtripathi** [comment]
added test

**Pull request merged**